mediaflow_proxy/extractors/dlhd.py

```
@@ -14,157 +15,262 @@ def __init__(self, request_headers: dict):
+            baseurl = await get_daddylive_base_url()
+            endpoints = ["stream/", "cast/", "player/", "watch/"]
+            last_exc = None
+            for endpoint in endpoints:
+                try:
+                    return await try_endpoint(baseurl, endpoint, channel_id)
+                except Exception as exc:
+                    last_exc = exc
+                    continue
+            raise ExtractorError(f"Extraction failed: {str(last_exc)}")
+        except Exception as e:
+            raise ExtractorError(f"Extraction failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)